x-pack/agent/pkg/fleetapi/round_trippers.go

```diff
@@ -48,7 +48,7 @@ func (r *FleetAuthRoundTripper) RoundTrip(req *http.Request) (*http.Response, er
 
 	req.Header.Set(key, prefix+r.apiKey)
 	resp, err := r.rt.RoundTrip(req)
-	if resp.StatusCode == http.StatusUnauthorized {
+	if err == nil && resp.StatusCode == http.StatusUnauthorized {
```

Comment:
oh it took me a while but i see, i found this not that readable. i think conventional 
```go
if err != nil {
  return resp, err
}

if resp.StatusCode ...
```

reads better. this confuses me and get me thinking can statusCode be unauthorized with error set, in which cases...